cromwell_tools/cromwell_tools.py

```diff
@@ -214,3 +226,86 @@ def read_local_file(path):
     with open(path) as f:
         contents = f.read()
     return contents
+
+
+def _content_checker(regex, content):
+    """
+    Helper function to check if a string is obeying the rule described by a regex string or not.
+
+    :param str regex: A regex string defines valid content.
+    :param str content: A string to be validated.
+
+    :return str: A string of error message if validation fails, or an empty string if validation succeeds.
+    """
+    try:
+        matched = re.fullmatch(regex, content)
+    except AttributeError:
+        matched = _fullmatch(regex, content)
+
+    if not matched:
+        return 'Invalid label: {0} did not match the regex {1}.\n'.format(content, regex)
+    else:
+        return ''
+
+
+def _length_checker(length, content):
+    """
+    Helper function to check if a string is shorter than expected length of not.
+
+    :param int length: Maximum length of an expected string.
+    :param str content: A string to be validated.
+
+    :return str: A string of error message if validation fails, or an empty string if validation succeeds.
+    """
+    if len(content) > length:
+        return 'Invalid label: {0} has {1} characters. The maximum is {2}.\n'.format(content, len(content), length)
+    else:
+        return ''
+
+
+def _fullmatch(regex, string, flags=0):
+    """
+    Backport Python 3.4's regular expression "fullmatch()" to Python 2 by emulating python-3.4 re.fullmatch().
+
+    If the whole string matches the regular expression pattern, return a corresponding match object.
+     Return None if the string does not match the pattern; note that this is different from a zero-length match.
+
+    :param str regex: A regex string.
+    :param str string: The string that you want to apply regex match to.
+    :param str/int flags: The expression’s behaviour can be modified by specifying a flags value. Values can be any of
```

Comment:
Normal convention is or = '|'

Reply:
Thanks, I was thinking that Sphinx might have trouble paring the "<space>or<space>", but `|`looks good to me!